### PR TITLE
DHFPROD-7316: Add pagination to source table in mapping

### PR DESCRIPTION
**Pull request by xnikhil08**
### Description
The relevant RTL test is added for pagination. No e2e is needed just to test pagination functionality.
UI changes are approved by @jbelonoj.

#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

File: marklogic-data-hub-central/ui/src/components/entities/mapping/mapping-step-detail/mapping-step-detail.tsx [hunk 1]
```diff
@@ -1295,6 +1295,11 @@ const MappingStepDetail: React.FC = () => {
     setOpenStepSettings(false);
   };
 
+  const paginationOptions = {
```
Comment (wooldridge) [inline marklogic-data-hub-central/ui/src/components/entities/mapping/mapping-step-detail/mapping-step-detail.tsx:1298]: @xnikhil08 can we add this information to a new config/mapping.config.js file?

Can we also get the tableColors (line 144) from there as well?

**Review by wooldridge — CHANGES_REQUESTED** [review]

File: marklogic-data-hub-central/ui/src/components/entities/mapping/mapping-step-detail/mapping-step-detail.tsx (hunk 1 above)
Comment (xnikhil08) [reply marklogic-data-hub-central/ui/src/components/entities/mapping/mapping-step-detail/mapping-step-detail.tsx]: @wooldridge - Done.

File: marklogic-data-hub-central/ui/src/components/entities/mapping/mapping-step-detail/mapping-step-detail.tsx [hunk 2]
```diff
@@ -26,6 +26,7 @@ import arrayIcon from "../../../../assets/icon_array.png";
 import relatedEntityIcon from "../../../../assets/icon_related_entities.png";
 import CustomPageHeader from "../../page-header/page-header";
 import {clearSessionStorageOnRefresh, getViewSettings, setViewSettings} from "../../../../util/user-context";
+import {paginationOptions, tableColors} from "../../../../config/mapping.config";
```
Comment (wooldridge) [inline marklogic-data-hub-central/ui/src/components/entities/mapping/mapping-step-detail/mapping-step-detail.tsx:29]: To display the colored tables correctly from the config file, you need to clone it, e.g.:

_in mapping.config.js_

export const mappingColors = ["#e4f1f4", "#f0f9f5",...

_in mapping-step-detail.tsx_

import {paginationOptions, mappingColors} from "../../../../config/mapping.config";
...
let tableColors = [...mappingColors];

**Review by wooldridge — CHANGES_REQUESTED** [review]

File: marklogic-data-hub-central/ui/src/components/entities/mapping/mapping-step-detail/mapping-step-detail.tsx (hunk 2 above)
Comment (xnikhil08) [reply marklogic-data-hub-central/ui/src/components/entities/mapping/mapping-step-detail/mapping-step-detail.tsx]: Fixed it.

**Review by wooldridge — APPROVED** [review]

**Review by devinnicholson — APPROVED** [review]

**Pull request merged**